plugins/CoreHome/angularjs/siteselector/siteselector-model.service.js

```diff
@@ -63,17 +64,31 @@
             }
 
             if (model.isLoading) {
-                model.currentRequest.abort();
+                if (model.currentRequest) {
+                    model.currentRequest.abort();
+                } else if (limitPromise) {
+                    limitPromise.abort();
+                    limitPromise = null;
+                }
             }
 
             model.isLoading = true;
 
-            model.currentRequest = piwikApi.fetch({
-                method: 'SitesManager.getPatternMatchSites',
-                pattern: term
-            });
+            if (!limitPromise) {
+                limitPromise = piwikApi.fetch({method: 'SitesManager.getNumWebsitesToDisplayPerPage'});
+            }
```

Comment:
I added it not as HTML attribute since we wanna get rid of our Controllers/Twig one day ;) and everything needs to be requested via API then. You're right it rather belongs to a JS controller or directive but put it here as it was easiest and it seems unlikely that we will reuse this model (eg `loadSite` also belongs in controller etc and in other components like All Websites Dashboard we needed to access the `MultiSites` API and not `SitesManager`). I will have a look to move it to controller or directive probably by the end of the week